README.md

```diff
@@ -30,6 +30,131 @@ JcatFile
+    Will the certificate be used for IPsec IKE operations? (y/N): N
+    Is this a TLS web server certificate? (y/N): N
+    Enter a dnsName of the subject of the certificate:
+    Enter an additional dnsName of the subject of the certificate:
+    Enter a URI of the subject of the certificate:
+    Enter the IP address of the subject of the certificate:
+    Enter the e-mail of the subject of the certificate: rhughes@hughski.com
+    Will the certificate be used for signing (required for TLS)? (Y/n): Y
+    Will the certificate be used for encryption (not required for TLS)? (Y/n): N
+    Will the certificate be used for data encryption? (y/N): N
+    Will the certificate be used to sign OCSP requests? (y/N): N
+    Will the certificate be used to sign code? (y/N): Y
+    Will the certificate be used for time stamping? (y/N): N
+    Will the certificate be used for email protection? (y/N): N
+    Will the certificate be used to sign other certificates? (Y/n): N
+    Will the certificate be used to sign CRLs? (y/N): N
+    Enter the URI of the CRL distribution point:
+
+Then we can actually use both files:
+
+    $ jcat-tool --appstream-id com.dell sign firmware.jcat firmware.bin test.pem test-privkey.pem
+    JcatFile:
+      Version:               0.1
+      JcatItem:
+        ID:                  firmware.bin
+        JcatBlob:
+          Kind:              pkcs7
+          Flags:             is-utf8
+          AppstreamId:       com.dell
+          Timestamp:         2020-03-05T12:16:30Z
+          Size:              0x373
+          Data:              -----BEGIN PKCS7-----
+                             MIICZwYJKoZIhvcNAQcCoIICWDCCAlQCAQExDTALBglghkgBZQMEAgEwCwYJKoZI
+                             ...
+                             8jggo0FbhDSs8frXhr1BHKBktOPKEbA3sETxlbHViYt6oldpi1uszV0kHA==
+                             -----END PKCS7-----
+
+Lets verify this new signature:
+
+    jcat-tool --appstream-id com.dell verify firmware.jcat
+    firmware.bin:
+        FAILED pkcs7: failed to verify data for C=UK,...: Public key signature verification has failed. [-89]
+        FAILED: Validation failed
+    Validation failed
+
+Ahh, of course; we need to tell Jcat to load our generated certificate:
+
+    jcat-tool --appstream-id com.dell verify firmware.jcat --public-key test.pem
+    firmware.bin:
+        PASSED pkcs7: C=UK,O=Hughski Limited,OU=Engineering,UID=rhughes,CN=Richard Hughes
+
+We can then check the result using
+
+    $ jcat-tool export firmware.jcat
+    Wrote ./firmware.bin-com.dell.p7b
+    $ certtool --p7-verify --infile firmware.bin-com.dell.p7b --load-data firmware.bin --load-certificate=test.pem
+    eContent Type: 1.2.840.113549.1.7.1
+    Signers:
+        Signer's issuer DN: C=UK,O=Hughski Limited,OU=Engineering,UID=rhughes,CN=Richard Hughes
+        Signer's serial: 4df758978d0601c6500ab6f266963916d8b7ab33
+        Signature Algorithm: RSA-SHA256
+        Signature status: ok
+
```

Comment:
So I think it's going to be especially important to have jcat generated a Windows EXE for this purpose so that more organizations can potentially integrate into infrastructure.

That being said, it might still not work on some infrastructure that the private key isn't ever exposed IIRC (like an HSA).

Reply:
> jcat generated a Windows EXE

Right, that's also why I'm just not bothered about GPG pubkey signed versions of this; getting GPG working in win32 would be a *nightmare*.

> like an HSA

Right, if you're using an HSA you're more than capable of using `import` on the command line. :)

Reply:
True :)